rholang/src/main/scala/coop/rchain/rholang/interpreter/util/Address.scala

```diff
@@ -0,0 +1,17 @@
+package coop.rchain.rholang.interpreter.util
+
+import java.nio.charset.StandardCharsets
+
+import coop.rchain.crypto.hash.Blake2b256
+import coop.rchain.rholang.interpreter.util.codec.Base58
+
+object Address {
```

Comment:
We also should have a 
`def parse(string: String): Option[Address]` that would:
1. check the `checksum`
2. take the last N bytes before the checksum, where N is our allowed key length, call it `pkHash`
3. treat the rest of the bytes on the beginning as the `prefix`
4. return `Address(prefix, pkHash, checksum)`

Reply:
We should have tests showing that `fromPublicKey` only produces `parse`-able addreses, and that various invalid input strings are rejected by `parse`.

Reply:
> We also should have a `def parse(string: String): Option[Address]` that would:
Agreed.

I'm currently working on a parse function with the signature `String => Either[String, Address]` that produces proper error messages depending on the

Reply:
> We should have tests showing that `fromPublicKey` only produces `parse`-able addreses, and that various invalid input strings are rejected by `parse`.
Aren't these generative tests from Base58Properties enough?

property("decode after encode returns the original bytes") 
property("decode a string should fail if it contains invalid Base58 chars")

Reply:
I misunderstood your comment. I'll address them by adding Address tests